### PR TITLE
Add functionality to parse the kernel config

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-kernel.c [hunk 1]
```diff
@@ -230,6 +230,141 @@ fu_kernel_reset_firmware_search_path(GError **error)
...
+#ifdef __linux__
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *fn = NULL;
+
+	g_return_val_if_fail(error == NULL || *error == NULL, NULL);
+
+	fn = fu_kernel_get_config_path(error);
+	if (fn == NULL)
+		return NULL;
```
Comment (superm1) [inline libfwupdplugin/fu-kernel.c:355]: how about using `/proc/config.gz` first? (IE `CONFIG_IKCONFIG`)

File: libfwupdplugin/fu-kernel.c [hunk 2]
```diff
@@ -230,6 +230,141 @@ fu_kernel_reset_firmware_search_path(GError **error)
...
+/**
+ * fu_kernel_parse_config:
+ * @buf: (not nullable): cmdline to parse
+ * @bufsz: size of @bufsz
+ *
+ * Parses all the kernel options into a hash table. Commented out options are not included.
+ *
+ * Returns: (transfer container) (element-type utf8 utf8): keys from the cmdline
+ *
+ * Since: 1.9.1
```
Comment (superm1) [inline libfwupdplugin/fu-kernel.c:277]: 1.9.6

File: libfwupdplugin/fu-kernel.c [hunk 3]
```diff
@@ -230,6 +230,141 @@ fu_kernel_reset_firmware_search_path(GError **error)
...
+	memset(&name_tmp, 0, sizeof(struct utsname));
+	if (uname(&name_tmp) < 0) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "failed to read kernel version");
+		return NULL;
+	}
+	config_fn = g_strdup_printf("config-%s", name_tmp.release);
+	return g_build_filename("/boot/", config_fn, NULL);
```
Comment (superm1) [inline libfwupdplugin/fu-kernel.c:323]: should this be a path lookup?

File: libfwupdplugin/fu-kernel.c (hunk 2 above)
Comment (hughsie) [reply libfwupdplugin/fu-kernel.c:277]: Fixed, thanks.

File: libfwupdplugin/fu-kernel.c (hunk 3 above)
Comment (hughsie) [reply libfwupdplugin/fu-kernel.c:323]: FU_PATH_KIND_BOOT perhaps?

Comment (superm1) [reply libfwupdplugin/fu-kernel.c:323]: Yah

Comment (hughsie) [reply libfwupdplugin/fu-kernel.c:323]: Maybe FU_PATH_KIND_HOSTFS_BOOT

File: libfwupdplugin/fu-kernel.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-kernel.c:355]: Added.

File: libfwupdplugin/fu-kernel.c (hunk 3 above)
Comment (hughsie) [reply libfwupdplugin/fu-kernel.c:323]: Done.

**Pull request merged**